Endpoint.swift

```diff
@@ -13,8 +13,18 @@ import Alamofire
 public enum EndpointSampleResponse {
     case Success(Int, NSData)
     case Error(Int?, NSError?, NSData?)
+    case Lazy(()-> EndpointSampleResponse)
+
+    func evaluate() -> EndpointSampleResponse {
+        switch self {
+        case Success, Error: return self
+        case Lazy(let lazy):
+            return lazy().evaluate()
```

Comment:
I like this, but the variable name didn't make it immediately obvious how this was working. What do you think about renaming `lazy` to `closure` or something similar?